pyiron_contrib/atomistics/meamfit/meamfit.py

```diff
@@ -0,0 +1,317 @@
+
+    @property
+    def potential_timings_dataframe(self):
+        return self._potential_timings_dataframe
+
+    @property
+    def potentials(self):
+        return list(self._potential_timings_dataframe.index)
+
+    @property
+    def potential_paths(self):
+        return [posixpath.join(self.working_directory, pot) for pot in self.potentials]
+
+
+    @property
+    def random_seed(self):
+        incar = self.input['SEED']
+        if incar == 'random':
+            self.input['SEED'] = random.randint(0, 100000)
+            incar = self.input['SEED']
+        return incar
+
+    @random_seed.setter
+    def random_seed(self, seed):
+        self.input['SEED'] = seed
+
+    def set_input_to_read_only(self):
+        """
+        This function enforces read-only mode for the input classes, but it has to be implement in the individual
+        classes.
+        """
+        self.input.read_only = True
+
+    def validate_ready_to_run(self):
+        if self._calculation_dataframe.empty:
+            raise ValueError()   
+          
+    def write_input(self):
+        """
+        Call routines that generate the codespecifc input files
+
+        Returns:
+
+        """
+        if self.input['SEED'] == 'random':
+            self.input['SEED'] = random.randint(0, 100000)
+        self.input.write_file(file_name="settings", cwd=self.working_directory)
+        self._write_calc_db(calculation_dataframe=self._calculation_dataframe,
+                            file_name="fitdbse",
+                            cwd=self.working_directory)
+        self._copy_vasprun_xml(cwd=self.working_directory)
+
+    def add_job_to_fitting(self, job_id, time_step_start=0, time_step_end=-1, time_step_delta=10, quantity='E0',
+                           weight=[1.0, 0.0, 0.0]):
+        """
+
+        Args:
+            job_id (int): job_id of the vasp MD job you want to use for potential fitting.
+            time_step_start (int): initial timestep - after equilibration
+            time_step_end (int): last timestep to use
+            time_step_delta (int): time step
+            quantity (str): the property in the vasprun file to be fit to, and can take the values ['E0', 'Free‐Energy', 'Force']
+                            ‘E0’: to the total energy (specifically the E0, sigma‐>0 value in the vasprun file);
+                            ‘Free‐Energy’: will fit to the free energy (the F value in the vasprun file);  
+                            ‘Force’: will fit to atomic forces.
+                             Note that only the first two letters are in fact read in by MEAMfit, so that it is sufficient to
+                             write ‘Fr’ or ‘Fo’ for the second and third cases respectively.  
+            weight (list): default is [1.0, 0.0, 0.0].
+            
+            More information on these parameters are available in the MeamFit user manual- https://www.scd.stfc.ac.uk/Pages/MEAMfit-v2.aspx
```

Comment:
I cannot find a manual under the given link.  Is it not public?

Reply:
The manual is not available publicly. In order to use the code users has to ask for permission individually from Andrew Ian Duff (andrew.duff@stfc.ac.uk)​​. Andrew send the code and manual to the user with permission to use the code.

Reply:
Ok, then let's just put a comment to that effect into the main docstring of the class.